Replace can-define with can-observable-*

chasenlehara opened the pull request:
This uses `can-observable-object` and `can-observable-array` as the defaults if no `ObjectType` or `ArrayType` are provided as `options`. The docs have also been updated to show the new packages.

Part of https://github.com/canjs/canjs/issues/5207

Closes https://github.com/canjs/can-realtime-rest-model/issues/13

Commit:
Replace can-define with can-observable-*

This uses `can-observable-object` and `can-observable-array` as the defaults if no `ObjectType` or `ArrayType` are provided as `options`. The docs have also been updated to show the new packages.

Part of https://github.com/canjs/canjs/issues/5207

Closes https://github.com/canjs/can-realtime-rest-model/issues/13

phillipskevin commented on can-realtime-rest-model.md:
```diff
@@ -218,64 +220,70 @@ The following code example uses `realtimeRestModel` to create a list of todos th
             <p>
                 <label>Name</label>
-                <input on:input:value:bind='todo.name'/>
+                <input on:input:value:bind="todo.name" />
```
This should be `this.todo.name`.

phillipskevin replied:
Same for the ones below.

phillipskevin commented on can-realtime-rest-model.md:
```diff
@@ -218,64 +220,70 @@ The following code example uses `realtimeRestModel` to create a list of todos th
+class TodoList extends StacheElement {
+    static view = `
+      <ul>
+          {{# if(todosPromise.isResolved) }}
```
This should be `this.todosPromise.isResolved`. Same thing for the one below.

phillipskevin commented on can-realtime-rest-model.md:
```diff
@@ -218,64 +220,70 @@ The following code example uses `realtimeRestModel` to create a list of todos th
             </p>
             <button disabled:from="todo.isSaving()">Create Todo</button>
-            {{# if(todo.isSaving()) }}Creating ....{{/ if}}
+            {{# if(todo.isSaving()) }}Creating…{{/ if}}
```
Need a space here. `{{/ if}}` -> `{{/ if }}`.

phillipskevin commented on can-realtime-rest-model.md:
```diff
@@ -218,64 +220,70 @@ The following code example uses `realtimeRestModel` to create a list of todos th
+                      <label>{{todo.name}}</label>
+                  </li>
+              {{/ for }}
+          {{/ if}}
```
Need a space here. `{{/ if}}` -> `{{/ if }}`.

phillipskevin commented on can-realtime-rest-model.md:
```diff
@@ -292,62 +300,70 @@ and shared. For example, if a `<todo-details>` widget loads the `id=5` todo, and
-    view: `
+class TodoDetails extends StacheElement {
+    static view = `
         Todo Name is: {{todoPromise.value.name}}
```
`todoPromise` -> `this.todoPromise`.

phillipskevin reviewed (changes requested):
This looks good. Just a few small style things.

Commit:
Update based on Kevin’s feedback

https://github.com/canjs/can-realtime-rest-model/pull/14#pullrequestreview-286280270

phillipskevin commented on can-realtime-rest-model.md:
```diff
@@ -314,7 +314,7 @@ Todo.connection = realtimeRestModel({
 class TodoDetails extends StacheElement {
     static view = `
-        Todo Name is: {{todoPromise.value.name}}
+        Todo name is: {{this.todoPromise.value.name}}
```
can you update the spacing on this one while we're here? `{{ this.todoPromise.value.name }}`.

phillipskevin reviewed (approved):
One more small thing. Feel free to merge after that.

Commit:
Minor fix based on Kevin’s feedback

https://github.com/canjs/can-realtime-rest-model/pull/14#pullrequestreview-286342372
